updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,167 @@
++
+[NOTE]
+====
+You can change `maxUnavailable` in a MCP to specify the percentage or the number of machines that can be updating at any given time. default is 1.
+====
++
+. Test if the applications are working as expected on the newly update MCP.
+
+. If you want to gohead with updating the rest of the MCPs then unpause rest of the MCPs one by one till all worker nodes are updated to the new version.
+
+== Steps to create a MCP
+
+. Get the list of worker nodes.
++
+[source,terminal]
+----
+$ oc get nodes | grep -v master
+----
+. Add the MCP name as label to the worker node and remove the `worker` label.
++
+[source,terminal]
+----
+$ oc label node <node name> node-role.kubernetes.io/<mcp name>=
+----
++
+[source,terminal]
+----
+$ oc label node <node name>  node-role.kubernetes.io/worker-
+----
+. Create a the machineconfig pool
++
+[source,yaml]
+----
+$ cat mcp_foo.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: mcp_foo | <1>
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,mcp_foo]} | <1>
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/mcp_foo: "" | <1>
+----
+<1> MCP name
++
+[source,terminal]
+----
+$ oc create -f mcp_foo.yaml
+----
++
+. To see the list of MCPs present in the cluster and their state
++
+[source,terminal]
+----
+$ oc get mcp
+----
+
+=== Pause a MCP
+
+Pausing a MCP will keep it from updating to the new OS version by the machine config operator.
+
+[source,terminal]
+----
+$ oc patch mcp/mcp_foo --patch '{"spec":{"paused":true}}' --type=merge
+----
+
+=== Unpause a MCP
+
+Unpausing will enable the nodes in the MCP to move to the new OS version and reboot (if required) it.
+
+[source,terminal]
+----
+$ oc patch mcp/mcp_foo --patch '{"spec":{"paused":false}}' --type=merge
+----
+
+== Steps to remove a node from a MCP
+
+A node must have a role to be properly functioning with in the OpenShift cluster.
+If you want to remove a node from a MCP, you should first relabel the node as a worker as it should be part of worker MCP if is not going to be part of any other MCP. Once it is labeled as worker only then proceed to unlabel it from the MCP.
+
+. Label it as worker
++
+[source,terminal]
+----
+$ oc label node <node name>  node-role.kubernetes.io/worker=
+----
++
+. Remove the MCP label
++
+[source,terminal]
+----
+$oc label node <node name> node-role.kubernetes.io/<mcp name>-
+----
++
+. The machine config operator is then going to reconcile the node to the worker pool configuration. Check the output of `oc get mcp` to make sure the worker pool is updated before going to the next step.
++
+[source,terminal]
+----
+$ oc get mcp
```

Comment:
For clarity, example output should list complete output from a testcluster that would also include fields like machinecount, readymachinecount, updatedmachinecount, etc. Also, it would be more clear to print result when worker pool has been completely updated i.e. UPDATED = true

Reply:
I agree, will put outputs here. I am not sure about calling out the details of the output. But open to suggestions.

Reply:
Added examples in https://github.com/openshift/openshift-docs/commit/3f24026e49caf7437eeff4b286bf49a8b9f141cc